### PR TITLE
[CIVIS] Add details on support for each CI about step spans

### PR DESCRIPTION
**Pull request by AdrianLC**
### What does this PR do? What is the motivation?

We had a support thread about a customer asking why this table says we have Steps for Azure Pipelines when we don't so we are trying to avoid the confusion by adding a new row to the feature matrix and a note on the tables about semantics.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/pipelines/_index.md [hunk 1]
```diff
@@ -151,6 +160,7 @@ If your CI provider is not supported, you can try setting up Pipeline Visibility
 | {{< ci-details title="Pipeline trace visualization" >}}Visualization of pipeline executions with associated tracing.{{< /ci-details >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |
 | {{< ci-details title="Running pipelines" >}}Identification of pipelines executions that are running with associated tracing.{{< /ci-details >}} | | {{< X >}} | | | {{< X >}} | | | | |
 | {{< ci-details title="Partial retries" >}}Identification of partial retries (for example, when only a subset of jobs were retried).{{< /ci-details >}} |  | {{< X >}} |  | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} | {{< X >}} |  {{< X >}} |
+| {{< ci-details title="Step spans" >}}Step level spans are available for more granular visibility.{{< /ci-details >}} | {{< X >}} _but handled as job spans_ | {{< X >}} |  |  |  |  | {{< X >}} |  |  |  {{< X >}} |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/pipelines/_index.md:163]:
```suggestion
| {{< ci-details title="Step spans" >}}Step level spans are available for more granular visibility.{{< /ci-details >}} | {{< X >}} _but presented as job spans_ | {{< X >}} |  |  |  |  | {{< X >}} |  |  |  {{< X >}} |
```

**Review by nikita-tkachenko-datadog — APPROVED** [review]

**Review by andrea-mosk — APPROVED** [review]

File: content/en/continuous_integration/pipelines/_index.md [hunk 2]
```diff
@@ -116,7 +123,9 @@ While the concept of a CI pipeline may vary depending on your provider, see how
 | Pipeline | Pipeline |
 | Stage | Stage |
 | Job | Job |
-| Step | Step |
+| Step* | Step |
+
+_\*Steps granularity is not available in Datadog_
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:128]:
```suggestion
_\*A pipeline's step granularity is not available in Datadog._
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 3]
```diff
@@ -106,7 +111,9 @@ While the concept of a CI pipeline may vary depending on your provider, see how
 | Pipeline | Build Chain |
 | Stage |  |
 | Job | Build |
-| Step |  |
+| Step* | Step |
+
+_\*Steps granularity is not available in Datadog_
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:116]:
```suggestion
_\*A pipeline's step granularity is not available in Datadog._
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 4]
```diff
@@ -85,18 +87,21 @@ While the concept of a CI pipeline may vary depending on your provider, see how
...
-
 | Datadog | Buildkite |
 |---|---|
 | Pipeline | Pipeline |
 | Stage |  |
 | Job | Job |
-| Step |  |
+| Step* | Step |
+
+_\*Steps granularity is not available in Datadog_
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:104]:
```suggestion
_\*A pipeline's step granularity is not available in Datadog._
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 5]
```diff
@@ -85,18 +87,21 @@ While the concept of a CI pipeline may vary depending on your provider, see how
 | Pipeline | Pipeline |
 | Stage | Workflow |
 | Job | Job |
-| Step | Step |
+| Step* | Step |
+
+_\*Steps granularity is not available in Datadog_
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:92]:
```suggestion
_\*A pipeline's step granularity is not available in Datadog._
```

File: content/en/continuous_integration/pipelines/_index.md [hunk 6]
```diff
@@ -65,7 +65,9 @@ While the concept of a CI pipeline may vary depending on your provider, see how
 | Pipeline | Pipeline |
 | Stage | Stage |
 | Job | Job |
-| Step |  |
+| Step* | Script |
+
+_\*Steps granularity is not available in Datadog_
```
Comment (alai97) [inline content/en/continuous_integration/pipelines/_index.md:70]:
```suggestion
_\*A pipeline's step granularity is not available in Datadog._
```

**Review by alai97 — COMMENTED** [review]
Looks good, some suggestions!

**Review by alai97 — APPROVED** [review]

**Pull request merged**